src/transformers/data/datasets/language_modeling.py

```diff
@@ -99,3 +101,119 @@ def __len__(self):
 
     def __getitem__(self, i) -> torch.Tensor:
         return torch.tensor(self.examples[i], dtype=torch.long)
+
+class AlbertTextDataset(Dataset):
+    """
+    Dataset for Albert Pretraining, prepare sentence pairs for SOP task
+    """
+
+    def __init__(self, tokenizer: PreTrainedTokenizer, file_dir: str, block_size: int, random_seed: int):
+        assert os.path.isdir(file_dir)
+        logger.info(f"Creating features from dataset file folder at {file_dir}")
+        self.examples = []
+        # TODO: randomness could apply a random seed, ex. rng = random.Random(random_seed)
+        for file_name in os.listdir(file_dir):
+            file_path = os.path.join(file_dir, file_name)
+            assert os.path.isfile(file_path)
+            with open(file_path, encoding="utf-8") as f:
+                document = [tokenizer.convert_tokens_to_ids(tokenizer.tokenize(line)) 
+                            for line in f.read().splitlines() if (len(line) > 0 and not line.isspace())]
+                examples = self.create_examples_from_document(document, block_size, tokenizer)
+                self.examples.extend(examples)
+
+    def create_examples_from_document(self, document, block_size, tokenizer, short_seq_prob=0.1):
+        """Creates examples for a single document."""
+
+        # Account for [CLS], [SEP], [SEP]
+        max_num_tokens = block_size - 3
```

Comment:
This can be tokenizer-agnostic and cleaner with the following
```suggestion
        # Account for special tokens
        max_num_tokens = block_size - tokenizer.num_special_tokens_to_add(pair=True)
```

Reply:
updated